typify-impl/src/value.rs

```diff
@@ -524,14 +551,160 @@ mod tests {
 
     #[test]
     fn test_value_numbers_and_string() {
-        let (type_space, type_id) = get_type::<(u32, i64, f64, String)>();
+        let (type_space, type_id) = get_type::<(u32, i64, f32, f64, String)>();
+        let type_entry = type_space.id_to_entry.get(&type_id).unwrap();
+
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!([0, 0, 0, 0, "zero"]))
+                .map(|x| x.to_string()),
+            Some(r#"(0_u32 , 0_i64 , 0_f32 , 0_f64 , "zero" . to_string ())"#.to_string()),
+        );
+    }
+
+    #[test]
+    fn test_value_validate_integers() {
+        let mut schema = schema_for!(u32);
+        let validation = Some(Box::new(NumberValidation {
+            multiple_of: Some(5.0),
+            minimum: Some(20.0),
+            maximum: Some(30.0),
+            exclusive_maximum: None,
+            exclusive_minimum: None,
+        }));
+        drop(schema.schema.number);
+        schema.schema.number = validation;
+
+        let type_name = type_name::<u32>();
+        let name = type_name
+            .rsplit_once("::")
+            .map_or(type_name, |split| split.1)
+            .to_string();
+
+        let mut type_space = TypeSpace::default();
+
+        // Convert all references
+        type_space
+            .add_ref_types(schema.definitions.clone())
+            .unwrap();
+        let type_id = if let Some(already_type_id) = type_space.ref_to_id.get(&name) {
+            already_type_id.clone()
+        } else {
+            type_space
+                .add_type_with_name(&schema.schema.into(), Some(name.to_string()))
+                .unwrap()
+        };
+
+        let type_entry = type_space.id_to_entry.get(&type_id).unwrap();
+
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(25))
+                .map(|x| x.to_string()),
+            Some("25_i8".to_string()),
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(20))
+                .map(|x| x.to_string()),
+            Some("20_i8".to_string()),
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(30))
+                .map(|x| x.to_string()),
+            Some("30_i8".to_string()),
```

Comment:
@ahl I'm a little confused on the intended behavior here. It seems to make sense that we would make it an `i8` given that the bounds fit within it, but we are changing it from the format that was given. This could be an issue in floats if you look at my other comment.